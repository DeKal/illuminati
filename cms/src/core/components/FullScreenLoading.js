import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import CircularProgress from '@material-ui/core/CircularProgress'

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    width: '100vw',
    height: '100vh',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    zIndex: theme.zIndex.modal,
    backgroundColor: 'rgba(0,0,0,0.7)'
  },
  bottom: {
    color: theme.colors.primaryBg,
    animationDuration: '550ms'
  }
}))

const FullScreenLoading = ({ isLoading }) => {
  const classes = useStyles()

  return (
    isLoading && (
      <div className={classes.root}>
        <CircularProgress
          size={52}
          thickness={4}
          className={classes.bottom}
          variant="indeterminate"
          disableShrink
        />
      </div>
    )
  )
}

export default FullScreenLoading

FullScreenLoading.propTypes = {
  isLoading: PropTypes.bool
}
