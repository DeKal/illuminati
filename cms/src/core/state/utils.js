export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
