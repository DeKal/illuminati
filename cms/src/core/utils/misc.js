import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url, isExact = false) => {
  if (url === '/' || pathName === '/' || isExact) {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}

export const arrayToAttributes = array =>
  array.reduce((object, attribute) => {
    return {
      ...object,
      [attribute]: true
    }
  }, {})

export const attributesToArray = obj =>
  Object.keys(obj).map(attribute => (obj[attribute] ? attribute : null))
