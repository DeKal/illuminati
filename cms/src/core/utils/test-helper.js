import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
