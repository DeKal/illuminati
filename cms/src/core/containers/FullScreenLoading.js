import { connect } from 'react-redux'
import FullScreenLoading from 'core/components/FullScreenLoading'
import { getIsAppLoading } from 'core/state/app/selectors'

const mapStateToProps = state => ({
  isLoading: getIsAppLoading(state)
})

export default connect(mapStateToProps)(FullScreenLoading)
