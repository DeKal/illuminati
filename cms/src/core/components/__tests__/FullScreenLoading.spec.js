import * as React from 'react'
import { mountWithTheme } from 'core/utils/test-helper'
import FullScreenLoading from '../FullScreenLoading'

describe('FullScreenLoading', () => {
  it('should render with no error', function() {
    const wrap = mountWithTheme(<FullScreenLoading isLoading />)
    expect(wrap).toMatchSnapshot()
  })

  it('should not render with no isLoading', function() {
    const wrap = mountWithTheme(<FullScreenLoading />)
    expect(wrap).toMatchSnapshot()
  })
})
