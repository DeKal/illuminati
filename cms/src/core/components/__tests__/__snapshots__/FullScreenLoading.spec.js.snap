// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenLoading should not render with no isLoading 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <FullScreenLoading />
</div>
`;

exports[`FullScreenLoading should render with no error 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <FullScreenLoading
    isLoading={true}
  >
    <div
      className="makeStyles-root-1"
    >
      <WithStyles(ForwardRef(CircularProgress))
        className="makeStyles-bottom-2"
        disableShrink={true}
        size={52}
        thickness={4}
        variant="indeterminate"
      >
        <ForwardRef(CircularProgress)
          className="makeStyles-bottom-2"
          classes={
            Object {
              "circle": "MuiCircularProgress-circle",
              "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
              "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
              "circleStatic": "MuiCircularProgress-circleStatic",
              "colorPrimary": "MuiCircularProgress-colorPrimary",
              "colorSecondary": "MuiCircularProgress-colorSecondary",
              "indeterminate": "MuiCircularProgress-indeterminate",
              "root": "MuiCircularProgress-root",
              "static": "MuiCircularProgress-static",
              "svg": "MuiCircularProgress-svg",
            }
          }
          disableShrink={true}
          size={52}
          thickness={4}
          variant="indeterminate"
        >
          <div
            className="MuiCircularProgress-root makeStyles-bottom-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "height": 52,
                "width": 52,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20}
                strokeWidth={4}
                style={Object {}}
              />
            </svg>
          </div>
        </ForwardRef(CircularProgress)>
      </WithStyles(ForwardRef(CircularProgress))>
    </div>
  </FullScreenLoading>
</div>
`;
